src/perform_step/bdf_perform_step.jl

```diff
@@ -868,28 +879,43 @@ function perform_step!(integrator,cache::QNDFCache,repeat_step=false)
       γₖ[i] = 1//1
     end
   end
-  γ = inv((1-κ)*γₖ[k])
-  nlsolver.γ = γ
+  β₀ = inv((1-κ)*γₖ[k])
+  α₀ = 1
+
   # precalculations
   ϕ = fill!(utilde, zero(eltype(u)))
   for i = 1:k
     @. ϕ += γₖ[i]*D[i]
   end
-  #@.. ϕ *= γ
-  lmul!(γ,ϕ)
-  tm = fill!(nlsolver.ztmp, zero(eltype(u)))
+  lmul!(β₀,ϕ)
+  tm = fill!(ztmp, zero(eltype(u)))
   for i = 1:k
     @. tm += D[i]
   end
-  @.. nlsolver.tmp = uprev + tm - ϕ
 
-  γdt = γ*dt
   markfirststage!(nlsolver)
   # initial guess
-  @.. nlsolver.z = (uprev + tm -nlsolver.tmp)*inv(γ)
+  @.. nlsolver.z = uprev + tm
+
+  mass_matrix = f.mass_matrix
+  
+  if mass_matrix === I
+    @.. tmp = (uprev + tm - ϕ)/ (dt * β₀)
+    
+  else
+    dz = nlsolver.cache.dz
```

Comment:
I am out of sync due to the MLH things, but we don't handle mass matrix intrinsically in nlsolve? tmp depending on mass matrix sounds spectical

Reply:
No, this is right.

Reply:
Alright 👍 I need to see the new nlsolver changes.